lib/setup_logging.dart

```diff
@@ -31,35 +34,14 @@ Future<void> setupLogging() async {
   final startupLogger = Logger("Startup");
   startupLogger.info("App starting, logging initialized.");
 
-  final packageInfo = await PackageInfo.fromPlatform();
-  final deviceInfo = DeviceInfoPlugin();
+  final metadata = MetaData();
+  await metadata.init();
 
-  startupLogger.info(
-      "This is ${packageInfo.appName} version ${packageInfo.version}+${packageInfo.buildNumber} (Signature '${packageInfo.buildSignature}'), installed via ${packageInfo.installerStore}.");
+  final packageInfo = metadata.appInfo;
+  final deviceInfoString = metadata.deviceInfo;
 
-  String deviceInfoString;
-  if (Platform.isAndroid) {
-    final androidInfo = await deviceInfo.androidInfo;
-    deviceInfoString =
-        "Android ${androidInfo.version.release} on ${androidInfo.model} (${androidInfo.product})";
-  } else if (Platform.isIOS) {
-    final iosInfo = await deviceInfo.iosInfo;
-    deviceInfoString = "${iosInfo.systemVersion} on ${iosInfo.model}";
-  } else if (Platform.isMacOS) {
-    final macosInfo = await deviceInfo.macOsInfo;
-    deviceInfoString =
-        "macOS ${macosInfo.majorVersion}.${macosInfo.minorVersion}.${macosInfo.patchVersion} on ${macosInfo.model}";
-  } else if (Platform.isLinux) {
-    final linuxInfo = await deviceInfo.linuxInfo;
-    deviceInfoString = "${linuxInfo.version} on ${linuxInfo.id}";
-  } else if (Platform.isWindows) {
-    final windowsInfo = await deviceInfo.windowsInfo;
-    deviceInfoString =
-        "Windows ${windowsInfo.displayVersion} on ${windowsInfo.deviceId}";
-  } else {
-    final webInfo = await deviceInfo.webBrowserInfo;
-    deviceInfoString =
-        "Web browser ${webInfo.userAgent} on ${webInfo.platform}";
-  }
-  startupLogger.info("Running on $deviceInfoString.");
+  startupLogger.info(
+      "This is ${packageInfo.version} version ${packageInfo.version}+${packageInfo.originallyInstalledVersion} (Signature '${packageInfo.installationDate}'), installed via ${packageInfo.updateHistory}.");
```

Comment:
Please don't change the information that is logged. The missing information (like `appName`, `buildNumber`, etc. should be added to the `AppInfo` class)